vite.config.js

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable spaced-comment */
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference
+/// <reference types="vitest" />
+
+import { flowPlugin, esbuildFlowPlugin } from "@bunchtogether/vite-plugin-flow";
+import { defineConfig } from "vite";
+
+const define = {
+  __DEBUG__: false,
+  __TEST__: true,
+  __WEB__: true,
+
+  __SDK_HOST__: true,
+  __PATH__: true,
+
+  __PAYPAL_DOMAIN__: true,
+  __PAYPAL_API_DOMAIN__: true,
+
+  __POST_ROBOT__: JSON.stringify({
+    __GLOBAL_KEY__: `__post_robot__`,
+    __AUTO_SETUP__: false,
+    __IE_POPUP_SUPPORT__: false,
+    __GLOBAL_MESSAGE_SUPPORT__: true,
+    __SCRIPT_NAMESPACE__: false,
+  }),
+  __PAYPAL_CHECKOUT__: JSON.stringify({
+    _MAJOR_VERSION__: "",
+    __MINOR_VERSION__: "",
+  }),
+};
+
+// eslint-disable-next-line import/no-default-export
+export default defineConfig({
+  define,
+  esbuild: {
+    define,
+  },
+  test: {
+    setupFiles: ["vitestSetup.js"],
+    environment: "jsdom",
+    clearMocks: true,
+    include: ["**/test/**/*.test.js"],
+    coverage: {
+      provider: "v8",
+      reportsDirectory: "./coverage",
+      include: ["src/**/*.js"],
+    },
+  },
+  optimizeDeps: {
+    esbuildOptions: {
+      plugins: [esbuildFlowPlugin()],
+    },
+  },
+  plugins: [flowPlugin({ exclude: "" })],
+});
```

Comment:
I had this thought:

> we should create a new grumbler-scripts-esque repo to contain all the vite config and npm scripts

but I think I prefer how it is now. e.g. it would take like 3 repo hops to identify where the karma dependency is coming from https://github.com/paypal/paypal-sdk-client/pull/181/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L14

just sharing the idea in case it's worth pursuing further (still on team independent repo config fwiw)

Reply:
we can just add those config to grumbler-scripts vs creating a new repo. it might not be a perfect tool (grumbler-scripts) but it would be consistent to keep shared configs there

Reply:
i always hated how grumbler-scripts was intended to hide dependencies as it does. thats also something we can pursue moving away from and I'd be in big support but lets stay incremental so we keep progress moving

Reply:
I'm intrigued by the idea, but do want to consider that out of the scope of this PR. A shared config could help a lot of the set up for sure.